src/models/feedDao.js

```diff
@@ -0,0 +1,227 @@
+const { AppDataSource } = require("./dataSource");
+
+    const getFeeds = async (userId, limit, offset) => {
+        const Feeds = await AppDataSource.query(`
+            SELECT
+                feeds.id AS id,
+                u.nickname AS userNickname,
+                b.image_url AS badge,
+                imgurl,
+                feeds.content,
+                CASE
+                    WHEN comment IS NOT NULL THEN comment
+                    ELSE 0
+                END AS comment,
+                is_challenge AS challenge, 
+                DATE_FORMAT(feeds.created_at, '%Y-%m-%d') AS created_at,
+                CASE
+                    WHEN feeds.user_id = ? THEN true
+                    ELSE false
+                END AS isMyPost
+            FROM 
+                feeds 
+            LEFT JOIN 
+                users u ON feeds.user_id = u.id
+            LEFT JOIN 
+                badges b ON u.badge_id = b.id
+            LEFT JOIN (
+                SELECT
+                    feed_id,
+                    COUNT(id) AS comment
+                FROM
+                    comments
+                GROUP BY feed_id
+                    ) AS comments ON feeds.id = comments.feed_id
+            LEFT JOIN (
+                SELECT
+                    feed_id,
+                JSON_ARRAYAGG(
+                    JSON_OBJECT(
+                        "id" , id,
+                        "url" , url
+                    )
+                ) AS imgurl
+                FROM
+                    feed_images
+                GROUP BY feed_id
+            ) AS feed_images ON feeds.id = feed_images.feed_id
+            ORDER BY
+                feeds.created_at DESC
+            LIMIT ? OFFSET ?
+            `,
+            [userId, limit, offset]
+        );
+        return Feeds;
+    }
+
+    // 총 피드 갯수
+    const getFeedCount = async () => {
+        const feedCount = await AppDataSource.query(`
+            SELECT COUNT(*) 
+            AS total 
+            FROM 
+                feeds
+        `);
+        return feedCount[0].total;
+      }
+
+    // 피드 작성
+    const addFeeds = async(userId, content, challenge, imageUrl) => {
+        // 중복 이미지
+        imageUrl = Array.from(new Set(imageUrl)).slice(0, 3);
+        // challenge 체크박스 값
+        const isChallenge = challenge === '0';
+
+        await AppDataSource.transaction(async (transactionManager) => {
+            const{ insertId:feedId } = await transactionManager.query(`
+                INSERT INTO feeds (
+                    user_id,
+                    content,
+                    is_challenge
+                ) VALUES (?, ?, ?)`,
+                [userId, content, isChallenge]
```

Comment:
```suggestion
                [userId, content, challenge]
```
혹시 isChallenge 변수 선언 안하고, 곧바로 challenge 넣으면 어떻게 되나요?

Reply:
0이나 1 선언한 값이 들어갑니다!
이 부분은 프론트랑 맞춰보고 어떻게 들어가는지 다시 한 번 검토해보겠습니다!